### PR TITLE
NXmx: read data_scale_factor

### PR DESCRIPTION
**Pull request by phyy-nx**
New parameter as of nexusformat/definitions#1343. This is not full support of the parameter or of data_offset, but it's one of the use cases needed, namely reading a single gain value

### CODE REVIEW

**Comment by phyy-nx** [comment]
Requires dials/nxmx#40

**Comment by phyy-nx** [comment]
Now includes a test using data from @biochem-fan (thanks!), derived from [CXI.DB #221](https://cxidb.org/id-221.html).  This is SACLA MPCCD data reformatted as NeXus NXmx, adding data_scale_factor as a parameter to account for a gain of 10.

@ndevenish this will need a new build of dials-data and of nxmx for these tests to pass.  Thanks!

**Comment by ndevenish** [comment]
Okay, this seems to be picking up the new nxmx now, errors of form:
```
>       detector = dxtbx.nexus.get_dxtbx_detector(instrument.detectors[0], wavelength)
E       TypeError: get_dxtbx_detector() missing 1 required positional argument: 'nxdata'

tests/nexus/test_build_dxtbx_models.py:275: TypeError
```

File: tests/nexus/test_mpccd_nexus.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+import dxtbx
+
+
+def test_mpccd_nexus_gain(dials_data):
+    """
+    Tests SACLA MPCCD image from CXI.DB 221
+    Includs parameter data_scale_factor, which accounts for a gain of 10
```
Comment (biochem-fan) [inline tests/nexus/test_mpccd_nexus.py:9]: Typo: "Includs"

File: src/dxtbx/nexus/__init__.py [hunk 2]
```diff
@@ -245,6 +245,7 @@ def get_dxtbx_scan(
 def get_dxtbx_detector(
     nxdetector: nxmx.NXdetector,
     wavelength: float,
+    nxdata: nxmx.NXdata = None,
```
Comment (ndevenish) [inline src/dxtbx/nexus/__init__.py:248]:
```suggestion
    nxdata: nxmx.NXdata | None = None,
```
No implicit-optional

**Comment by ndevenish** [comment]
Otherwise, I think this is ready-to-go?

**Comment by phyy-nx** [comment]
Great, thanks!

**Pull request merged**